### PR TITLE
Feature/apollo update

### PR DESCRIPTION
**Pull request by DanijelCBS**
- Added loan_to_value parameter in borrow extrinsic to make loan to value flexible not fixed
- Implemented add_collateral extrinsic for users to add collateral and strengthen their borrowing position
- Added tests for those cases

### CODE REVIEW

File: pallets/apollo-platform/src/weights.rs [hunk 1]
```diff
@@ -532,4 +548,19 @@ impl WeightInfo for () {
...
+	fn add_collateral() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `1210`
+		//  Estimated: `13530`
+		// Minimum execution time: 21_791 nanoseconds.
+		Weight::from_parts(22_489_000, 13530)
+			.saturating_add(RocksDbWeight::get().reads(4))
+			.saturating_add(RocksDbWeight::get().writes(3))
+	}
 }
```
Comment (Alexey-N-Chernyshov) [inline pallets/apollo-platform/src/weights.rs:566]:
```suggestion
}

```

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Comment by Tieumsan** [comment]
> * Added loan_to_value parameter in borrow extrinsic to make loan to value flexible not fixed

@DanijelCBS can you help me understand why you want the Loan To Value be flexible please ?

**Review by vovac12 — APPROVED** [review]

**Comment by DanijelCBS** [comment]
> > * Added loan_to_value parameter in borrow extrinsic to make loan to value flexible not fixed
> 
> @DanijelCBS can you help me understand why you want the Loan To Value be flexible please ?

The Loan to Value (“LTV”) ratio defines the maximum amount of assets that can be borrowed with a specific collateral. It is expressed as a percentage (e.g., at LTV=75%, for every 1 ETH worth of collateral, borrowers will be able to borrow 0.75 ETH worth of the corresponding currency.

Each pool has a predefined loan-to-value (LTV) ratio, which currently restricts users to borrowing based solely on this fixed ratio. This approach presents drawbacks since the predefined LTV represents the maximum allowable for the pool, potentially resulting in a lower health factor and increased liquidation risk. Instead, the defined LTV should serve as a ceiling, allowing users flexibility to choose their preferred LTV. Opting for a lower LTV would result in reduced borrowing but also lower liquidation risk and a higher health factor.

**Comment by DanijelCBS** [comment]
> > > * Added loan_to_value parameter in borrow extrinsic to make loan to value flexible not fixed
> > 
> > 
> > @DanijelCBS can you help me understand why you want the Loan To Value be flexible please ?
> 
> The Loan to Value (“LTV”) ratio defines the maximum amount of assets that can be borrowed with a specific collateral. It is expressed as a percentage (e.g., at LTV=75%, for every 1 ETH worth of collateral, borrowers will be able to borrow 0.75 ETH worth of the corresponding currency.
> 
> Each pool has a predefined loan-to-value (LTV) ratio, which currently restricts users to borrowing based solely on this fixed ratio. This approach presents drawbacks since the predefined LTV represents the maximum allowable for the pool, potentially resulting in a lower health factor and increased liquidation risk. Instead, the defined LTV should serve as a ceiling, allowing users flexibility to choose their preferred LTV. Opting for a lower LTV would result in reduced borrowing but also lower liquidation risk and a higher health factor.

At the moment when user borrows it is always the riskiest borrow because LTV is fixed to the max LTV for the pool.

**Comment by Tieumsan** [comment]
@DanijelCBS that's clear, thanks

**Pull request merged**